projboard/tests/test_signup.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+from projboard.forms import NewUserForm
+from projboard.models.user import User
+from conftest import (
+                    VALID_USER,
+                    VALID_EMAIL,
+                    VALID_PASSWORD,
+                    INVALID_EMAIL,
+                    INVALID_PASSWORD_MISMATCH,
+                    MISMATCH_MESSAGE_ERROR,
+                    VALID_NAME,
+                    SIGNUP_PATH
+                    )
+
+
+@pytest.mark.django_db
+def test_valid_form():
+    form_data = {
+        'email': VALID_EMAIL,
+        'name': VALID_NAME,
+        'nickname': VALID_USER,
+        'password': VALID_PASSWORD,
+        'password_confirm': VALID_PASSWORD,
+    }
+    form = NewUserForm(data=form_data)
+    if form.is_valid():
+        assert True
+    else:
+        assert False
+
+
+@pytest.mark.django_db
+def test_invalid_form():
+    form_data = {
+        'email': INVALID_EMAIL,
+        'name': VALID_NAME,
+        'nickname': VALID_USER,
+        'password': VALID_PASSWORD,
+        'password_confirm': VALID_PASSWORD,
+    }
+    form = NewUserForm(data=form_data)
+    if form.is_valid():
+        assert False
+    else:
+        assert True
+
+
+@pytest.mark.django_db
+def test_valid_registration(client):
+    valid_data = {
+        'email': VALID_EMAIL,
+        'name': VALID_NAME,
+        'nickname': VALID_USER,
+        'password': VALID_PASSWORD,
+        'password_confirm': VALID_PASSWORD,
+        }
+    response = client.post(SIGNUP_PATH, data=valid_data)
+    # ASSERTION TO SHOW THAT SIGNUP SWITCHED TO HOMEPAGE (LOGIN PAGE LATER)
+    assert response.status_code == 302
+    assert User.get_user_by_nickname(VALID_USER)
+    # THIS ASSERT IS SETTED FOR HOMEPAGE BECAUSE /LOGIN ARENT READY YET
+    assert response.url == '/'
```

Comment:
Correct, If you receive 302 
You could query again with client that you receive the correct page when you request

client.get("/")
And verify you are in the correct template

Reply:
You should verify the you continue to the right place after 302 reponse
It is another client client call

Reply:
@Yarboa Got it, please check 01b8113

Reply:
@Yarboa And rebased to #96

Reply:
Almost there, please check the template received, assert you are in the right place

Reply:
@Yarboa Done